packages/block-library/src/navigation/index.php

```diff
@@ -1383,7 +1383,7 @@ function block_core_navigation_get_most_recently_published_navigation() {
  * @param WP_Post $post         `wp_navigation` post object corresponding to the block.
  * @return string Serialized inner blocks in mock Navigation block wrapper, with hooked blocks inserted, if any.
  */
-function block_core_navigation_insert_hooked_blocks( $inner_blocks, $post ) {
+function block_core_navigation_insert_hooked_blocks( $inner_blocks, $post, $callback = 'insert_hooked_blocks' ) {
```

Comment:
The function name (`block_core_navigation_insert_hooked_blocks) isn't quite accurate anymore since it can now either insert hooked blocks, or set the `ignoredHookedBlocks` metadata.

Furthermore, requiring the callback name as an arg here is also possibly exposing too many internals.

Not sure if we want two separate functions (maybe using the same underlying helper, just with different callbacks?), e.g. `block_core_navigation_insert_hooked_blocks` and `block_core_navigation_set_ignored_hooked_blocks_metadata` or so.

Reply:
Done in https://github.com/WordPress/gutenberg/pull/59021/commits/43a927692aa4209688fd0c26d958d264ab182f18